sr_data_visualization/src/sr_data_visualization/data_plot.py

```diff
@@ -115,23 +143,93 @@ def show_trace(self, trace_name):
+            trace.latest_value = 0.0
+            trace.plot.setData(self.x_data, trace.data)
+        self.replot()
+
 
 class JointStatesDataPlot(GenericDataPlot):
     def __init__(self, joint_name, topic_name, topic_type):
         super().__init__(joint_name, topic_name, topic_type, start_plotting=True)
 
+        self.side = self.joint_name[0:2]
+
     def create_traces(self):
+        self.buffers_left = [DataBuffer(self.x_data.shape),
+                             DataBuffer(self.x_data.shape),
+                             DataBuffer(self.x_data.shape)]
+
+        self.buffers_right = [DataBuffer(self.x_data.shape),
+                              DataBuffer(self.x_data.shape),
+                              DataBuffer(self.x_data.shape)]
+
         self.traces = [Trace("Position", Qt.red, self.x_data),
                        Trace("Effort", Qt.blue, self.x_data),
                        Trace("Velocity", Qt.green, self.x_data)]
 
+    def plot_data(self, plot, side, new_sub=True):
+        self.side = side
+
+        if new_sub:
+
+            self.clear_data()
+
+            if plot:
+                self._subscriber = rospy.Subscriber(self._topic_name, self._topic_type,
+                                                    self.callback, queue_size=1)
+                if self.timer is None:
+                    self.initialize_and_start_timer()
+                else:
+                    self.timer.start()
+            elif self._subscriber is not None:
+                self._subscriber.unregister()
+                self.timer.stop()
+                self.clear_data()
+
     def callback(self, data):
         for name, position, velocity, effort in zip(data.name, data.position,
                                                     data.velocity, data.effort):
-            if name == self.joint_name:
-                self.traces[0].latest_value = position
-                self.traces[1].latest_value = effort
-                self.traces[2].latest_value = velocity
+            if name[2:] in self.joint_name[2:]:
+                if 'lh' in name:
+                    self.buffers_left[0].introduce(position)
+                    self.buffers_left[1].introduce(effort)
+                    self.buffers_left[2].introduce(velocity)
+                if 'rh' in name:
+                    self.buffers_right[0].introduce(position)
+                    self.buffers_right[1].introduce(effort)
+                    self.buffers_right[2].introduce(velocity)
+
+    def timerEvent(self):
+        # Data moves from left to right:
+        # Shift data array right and assign new value data[0]
+
+        for trace, buffer_left, buffer_right in zip(self.traces, self.buffers_left, self.buffers_right):
+            buffer_left.shift()
+            buffer_right.shift()
+
+            if self.side == 'lh':
+                trace.data = buffer_left.data
+            if self.side == 'rh':
+                trace.data = buffer_right.data
+
```

Comment:
If the above comment (regarding separate buffer makes sense) then this would require only one buffer